contracts/SnappAuction.sol

```diff
@@ -149,34 +151,41 @@ contract SnappAuction is SnappBase {
         emit StandingSellOrderBatch(currentBatchIndex, accountId, buyTokens, sellTokens, buyAmounts, sellAmounts);
     }
 
-    function placeSellOrder(
-        uint8 buyToken,
-        uint8 sellToken,
-        uint128 buyAmount,
-        uint128 sellAmount
-    ) public onlyRegistered() {
-
-        if (
-            auctionIndex == MAX_UINT ||
-            auctions[auctionIndex].size == maxUnreservedOrderCount() ||
-            block.timestamp > (auctions[auctionIndex].creationTimestamp + 3 minutes)
-        ) {
-            createNewPendingBatch();
-        }
-
-        // Update Auction Hash based on request
+    function placeSellOrders(bytes memory packedOrders) public onlyRegistered() {
+        // Note that this could result failure of all orders if even one fails.
+        require(packedOrders.length % 26 == 0, "Each order should be packed in 26 bytes!");
+        // TODO - use ECRecover from signature contained in first 65 bytes of packedOrder
         uint16 accountId = publicKeyToAccountMap(msg.sender);
-        bytes32 nextAuctionHash = sha256(
-            abi.encodePacked(
-                auctions[auctionIndex].shaHash,
-                encodeOrder(accountId, buyToken, sellToken, buyAmount, sellAmount)
-            )
-        );
-        auctions[auctionIndex].shaHash = nextAuctionHash;
-
-        emit SellOrder(auctionIndex, auctions[auctionIndex].size, accountId, buyToken, sellToken, buyAmount, sellAmount);
-        // Only increment size after event (so it is emitted as an index)
-        auctions[auctionIndex].size++;
+        bytes memory orderData;
+
+        for (uint i = 0; i < packedOrders.length / 26; i++) {
+            orderData = packedOrders.slice(26*i, 26);
+
+            uint8 buyToken = BytesLib.toUint8(orderData, 0);
+            uint8 sellToken = BytesLib.toUint8(orderData, 1);
+
+            uint96 buyAmount;
+            assembly {  // solhint-disable no-inline-assembly
+                buyAmount := mload(add(add(orderData, 0xc), 2))
+            }
+            uint96 sellAmount;
+            assembly {  // solhint-disable no-inline-assembly
+                sellAmount := mload(add(add(orderData, 0xc), 14))
+            }
+            orderBatchUpdate();  // Could this be done more efficiently?
+            bytes32 nextAuctionHash = sha256(
+                abi.encodePacked(
```

Comment:
Why do we here an encodePacked? Is there a risk if a sha hash starts with a zero?

Reply:
I don't understand your question. We have always been using `encodePacked` and I was just continuing to do so.

Reply:
Yeah, we have always been using encodePacked, 
I was wondering, whether there is a risk in case we are getting leading zeros in the sha hash and in the id. In this case the backend infrastructure would not be aware of where to truncate. I think it could cause problems and we might want to switch to encode only.
However, it might all work fine, as every string starts with 0x anyways and then there will not be any truncating.  But then we can use encode anyways, instead of encodePacked

Reply:
Perhaps, this would be better in the design discussion. I would love to learn more about this and see some examples/potential points of failure.